### PR TITLE
add broken test that fails because floats are first parsed as doubles

### PR DESCRIPTION
**Pull request by pjfanning**
test that demos the issue in https://github.com/FasterXML/jackson-core/issues/755

The array that is read in has the same first 4 elements that are used in the other array tests that are already in this class.

Something about the larger array (it has 1000 extra elements) leads to the numbers being parsed as doubles and later cast as floats (and this casting gives the wrong results). The existing tests somehow avoid this - parsing the numbers directly using Float.parseFloat.

### CODE REVIEW

File: src/test/java/com/fasterxml/jackson/databind/deser/TestFloats.java [hunk 1]
```diff
@@ -50,6 +54,28 @@ public void testArrayOfFloatPrimitives() throws Exception
         assertEquals("1.4E-45", Float.toString(floats[3])); //this assertion fails unless toString is used
     }
 
+    public void testBigArrayOfFloatPrimitives() throws Exception {
+        StringBuilder sb = new StringBuilder(1024);
+        ObjectMapper mapper = new ObjectMapper();
+        try (
+                InputStream stream = TestFloats.class.getResourceAsStream("/data/float-array.txt");
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/databind/deser/TestFloats.java:61]: Why not simply pass this `InputStream` to `ObjectMapper`? Or does that not exhibit the issue?

**Comment by cowtowncoder** [comment]
Size of the content is interesting -- it suggests to me there is a buffer condition. Code paths do vary a bit when a token (like float value) is on buffer boundary. Should not really change behavior, and at databind level there is no difference.

One thing that'd be useful, I think, would be to maybe try out streaming-level tests that uses `parser.getText()` (which can be used with number tokens too) to see if the issue is with:

1. Actual tokenization (parsing) -- if so, text value would be corrupt (missing/extra digit most likely)
2. Coercion -- maybe "happy path" differs somehow -- so there'd be double->float coercion (value almost correct but not quite)

Based on failure, which of above seems more likely?

File: src/test/java/com/fasterxml/jackson/databind/deser/TestFloats.java (hunk 1 above)
Comment (pjfanning) [reply src/test/java/com/fasterxml/jackson/databind/deser/TestFloats.java]: made the requested change - does not affect the test

**Comment by pjfanning** [comment]
> Size of the content is interesting -- it suggests to me there is a buffer condition. Code paths do vary a bit when a token (like float value) is on buffer boundary. Should not really change behavior, and at databind level there is no difference.
> 
> One thing that'd be useful, I think, would be to maybe try out streaming-level tests that uses `parser.getText()` (which can be used with number tokens too) to see if the issue is with:
> 
> 1. Actual tokenization (parsing) -- if so, text value would be corrupt (missing/extra digit most likely)
> 2. Coercion -- maybe "happy path" differs somehow -- so there'd be double->float coercion (value almost correct but not quite)
> 
> Based on failure, which of above seems more likely?

https://github.com/FasterXML/jackson-core/pull/757 seems to fix the issue - parseSlowFloat call happens when the bigger array is tested while somehow the tests with smaller arrays avoid parseSlowFloat

**Comment by cowtowncoder** [comment]
@pjfanning I wonder if my hunch was incorrect, just because the problem occurs so early (on index 3?). Well, I hope to find time today to look into this -- thank you for digging deep here!

**Review by cowtowncoder — APPROVED** [review]

**Pull request merged**